src/enzyme_ad/jax/Passes/LLVMToAffineAccess.cpp

```diff
@@ -1621,6 +1621,8 @@ struct SimplifyDeadAlloc : public OpRewritePattern<T> {
       }
 
       if (auto cpy = dyn_cast<enzymexla::MemcpyOp>(op)) {
+        if (cpy.getSource() == alloc->getResult(0))
```

Comment:
we should delete the target one

Reply:
and do you mind adding a quick test for this to avoid it recurring (and especially the threadid one)